pkg/asset/cluster/aws/aws.go

```diff
@@ -41,36 +40,32 @@ func PreTerraform(ctx context.Context, clusterID string, installConfig *installc
 
 	publicSubnets, err := installConfig.AWS.PublicSubnets(ctx)
 
-	arns := make([]string, 0, len(privateSubnets)+len(publicSubnets))
+	arns := make([]*string, 0, len(privateSubnets)+len(publicSubnets))
 	for _, subnet := range privateSubnets {
-		arns = append(arns, subnet.ARN)
+		arns = append(arns, aws.String(subnet.ARN))
 	}
 	for _, subnet := range publicSubnets {
-		arns = append(arns, subnet.ARN)
+		arns = append(arns, aws.String(subnet.ARN))
 	}
 
 	session, err := installConfig.AWS.Session(ctx)
 	if err != nil {
 		return err
 	}
 
-	request := &resourcegroupstaggingapi.TagResourcesInput{
-		Tags: map[string]*string{
-			fmt.Sprintf("kubernetes.io/cluster/%s", clusterID): aws.String("shared"),
+	tags := []*ec2.Tag{
+		{
+			Key:   aws.String(fmt.Sprintf("kubernetes.io/cluster/%s", clusterID)),
+			Value: aws.String("shared"),
 		},
 	}
+	client := ec2.New(session, aws.NewConfig().WithRegion(installConfig.Config.Platform.AWS.Region))
 
-	tagClient := resourcegroupstaggingapi.New(session, aws.NewConfig().WithRegion(installConfig.Config.Platform.AWS.Region))
-	for i := 0; i < len(arns); i += 20 {
-		request.ResourceARNList = make([]*string, 0, 20)
-		for j := 0; i+j < len(arns) && j < 20; j++ {
-			logrus.Debugf("Tagging %s with kubernetes.io/cluster/%s: shared", arns[i+j], clusterID)
-			request.ResourceARNList = append(request.ResourceARNList, aws.String(arns[i+j]))
-		}
-		_, err = tagClient.TagResourcesWithContext(ctx, request)
-		if err != nil {
-			return err
-		}
+	if _, err = client.CreateTagsWithContext(ctx, &ec2.CreateTagsInput{
+		Resources: arns,
```

Comment:
from CI:
```
ARNs are not supported. Please specify a resourceId.
```

Reply:
I think you want `SubnetId` from the `ec2.Subnet` rather than `SubnetArn`.